test/service/test_response.py

```diff
@@ -1772,6 +1772,61 @@ def test_pagination_search_after_search_before(self):
         self.assertEqual(second_page_previous['search_before'], 'null')
         self.assertEqual(second_page_previous['search_before_uid'], 'doc#308eea51-d14b-4036-8cd1-cfd81d7532c3')
 
+    def test_filter_by_publicationTitle(self):
+        expected_filtered_results = [
+            (
+                'Systematic comparative analysis of single cell RNA-sequencing methods',
+                [
+                    {
+                        'bundleUuid': 'ffac201f-4b1c-4455-bd58-19c1a9e863b4',
+                        'bundleVersion': '2019-10-09T170735.528600Z'
+                    }
+                ]
+            ),
+            (
+                'Palantir characterizes cell fate continuities in human hematopoiesis',
+                ()
+            )
+        ]
+        for publication_title, bundle_hits in expected_filtered_results:
+            with self.subTest(publication_title=publication_title):
+                url = furl(url=self.base_url, path='index/bundles').url
+                filters_parameter = {
+                    'publicationTitle': {
+                        'is': [publication_title]
+                    }
+                }
+                publication_term_facet = {
+                    'terms': [
+                        {
+                            'term': None,
+                            'count': 4
+                        },
+                        {
+                            'term': 'A title of a publication goes here.',
+                            'count': 1
+                        },
+                        {
+                            'term': 'Single-Cell Analysis of Human Pancreas Reveals Transcriptional '
+                                    'Signatures of Aging and Somatic Mutation Patterns.',
+                            'count': 1
+                        },
+                        {
+                            'term': 'Systematic comparative analysis of single cell RNA-sequencing methods',
+                            'count': 1
+                        }
+                    ],
+                    'total': 7,
+                    'type': 'terms'
+                }
+                response = requests.get(url=url,
+                                        params={"filters": json.dumps(filters_parameter)}).json()
+
+            self.assertEqual(response['termFacets']['publicationTitle'], publication_term_facet)
+            self.assertEqual(len(response['hits']), len(bundle_hits))
+            if len(bundle_hits) > 0:
+                self.assertEqual(one(response['hits'])['bundles'], bundle_hits)
```

Comment:
This looks fishy. In /index/bundles, hits.bundles is always a singleton but you put the `one` call around hits.

Reply:
There is only one bundle that has the publicationTitle `Systematic comparative analysis of single cell RNA-sequencing methods`, this was being used to enforce that only one hit was returned.

Reply:
If there were two bundles in the project with that title, you would need two entries in the expectation but the `one` call would fail. So this pretends to be variable in the number of hits by placing the expected bundles in the `cases` literal but the call to `one` torpedoes that by hard-coding the number of hits to 1. The thing that invariant is the number of bundle inner entities in each hit, not the number of hits that come back from the query. That one depends on the canned bundles the test indexes and the title in the `cases` literal.